conans/client/graph/graph_binaries.py

```diff
@@ -291,17 +291,15 @@ def _compute_package_id(self, node, default_package_id_mode, default_python_requ
         # A bit risky to be done now
         conanfile = node.conanfile
         neighbors = node.neighbors()
+
+        direct_reqs, indirect_reqs = self.package_id_transitive_reqs(node)
+
```

Comment:
I'd add my comment here, everything in this `if` block is there to reproduce a bug-as-a-feature to keep backward compatibility

```suggestion

        # FIXME: Conan v2.0 This is introducing a bug for backwards compatibility, it will add
        #   only the requirements available in the 'neighbour.info' object, not all the closure
```